diff --git a/.gitmodules b/.gitmodules
index 98dc79e..02f10d6 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,8 +1,3 @@
-[submodule "godot-cpp"]
-	path = godot-cpp
-	url = https://github.com/godotengine/godot-cpp.git
-	branch = 4.1
-	
 [submodule "src/thirdparty/tracy"]
 	path = src/thirdparty/tracy
 	url = https://github.com/wolfpld/tracy.git
diff --git a/godot-cpp b/godot-cpp
deleted file mode 160000
index a4f9d22..0000000
--- a/godot-cpp
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit a4f9d22ac54e05ca6a41d54a59afd7555cb7d292
